main.tf

```diff
@@ -15,8 +15,22 @@ locals {
   )
   aws_policy_prefix = format("arn:%s:iam::aws:policy", join("", data.aws_partition.current.*.partition))
 
+  node_labels = [
+    for item in keys(var.kubernetes_labels):
+      join("=", [item, lookup(var.kubernetes_labels, item)])
+  ]
+
   userdata_vars = {
-    before_cluster_joining_userdata = var.before_cluster_joining_userdata
+    before_cluster_joining_userdata = var.before_cluster_joining_userdata,
+    kubelet_extra_args = replace(join(" ", var.kubelet_extra_args), "'", ""),
+    node_taints = replace(join(",", var.node_taints), "'", ""),
+    node_labels = replace(join(",", local.node_labels),"'", "")
```

Comment:
I do not like the idea of blindly stripping quotes out of values. What if they have backslash escapes, for example?